### PR TITLE
Feature/epc qr code generator

### PR DESCRIPTION
**Pull request by julianzimmermann**
- extracted qr code generator stuff to new package
- use qr code generator package
- refactoring

### CODE REVIEW

File: bundles/payment-epc-qr-code/composer.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   ],
   "require": {
     "php": ">=7.3",
-    "endroid/qr-code": "^4.1.0",
+    "fond-of-oryx/qr-code-generator": "^dev-master",
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/composer.json:13]: Use correct version constraint.

File: bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php [hunk 2]
```diff
@@ -15,56 +16,92 @@ class PaymentEpcQrCodeConfig extends AbstractBundleConfig
     /**
      * @return string
      */
-    public function getEncoding(): string
+    public function getFormat(): string
     {
-        return $this->get(PaymentEpcQrCodeConstants::QR_CODE_ENCODING, 'UTF-8');
+        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_QR_CODE_FORMAT, 'png');
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php:21]: Use const for default value.

File: bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php [hunk 3]
```diff
@@ -11,64 +11,64 @@ class PaymentEpcQrCodeConfig extends AbstractBundleConfig
...
     }
 
     /**
      * @return string
      */
-    public function getVersion(): string
+    public function getEpcDataVersion(): string
     {
-        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_VERSION, '002');
+        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_DATA_VERSION, '002');
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php:40]: Use const for default value.

File: bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php [hunk 4]
```diff
@@ -11,64 +11,64 @@ class PaymentEpcQrCodeConfig extends AbstractBundleConfig
...
     }
 
     /**
      * @return string
      */
-    public function getServiceTag(): string
+    public function getEpcDataServiceTag(): string
     {
-        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_SERVICE_TAG, 'BCD');
+        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_DATA_SERVICE_TAG, 'BCD');
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php:48]: Use const for default value.

File: bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php [hunk 5]
```diff
@@ -11,64 +11,64 @@ class PaymentEpcQrCodeConfig extends AbstractBundleConfig
...
     }
 
     /**
      * @return string
      */
-    public function getEncoding(): string
+    public function getEpcDataEncoding(): string
     {
-        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_ENCODING, 'UTF-8');
+        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_DATA_ENCODING, 'UTF-8');
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php:56]: Use const for default value.

File: bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php [hunk 6]
```diff
@@ -11,64 +11,64 @@ class PaymentEpcQrCodeConfig extends AbstractBundleConfig
...
     }
 
     /**
      * @return string
      */
-    public function getType(): string
+    public function getEpcDataType(): string
     {
-        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_TYPE, 'SCT');
+        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_DATA_TYPE, 'SCT');
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php:64]: Use const for default value.

File: bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php [hunk 7]
```diff
@@ -11,64 +11,64 @@ class PaymentEpcQrCodeConfig extends AbstractBundleConfig
...
     }
 
     /**
      * @return string
      */
-    public function getPurpose(): string
+    public function getEpcDataPurpose(): string
     {
-        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_PURPOSE, '');
+        return $this->get(PaymentEpcQrCodeConstants::EPC_QR_CODE_DATA_PURPOSE, '');
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode/PaymentEpcQrCodeConfig.php:72]: Use const for default value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/Dependency/Wrapper/WriterCollection.php [hunk 8]
```diff
@@ -33,13 +34,26 @@ public function __construct(array $writerBag)
...
     {
+        $self = $this;
         $this->writer = [
-            'png' => $this->createPngWriter(),
-            'svg' => $this->createSvgWriter(),
-            'binary' => $this->createBinaryWriter(),
-            'eps' => $this->createEpsWriter(),
-            'pdf' => $this->createPdfWriter(),
-            'debug' => $this->createDebugWriter(),
+            'png' => static function () use ($self) {
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/Dependency/Wrapper/WriterCollection.php:39]: Use const for array keys.

File: codeception.yml [hunk 9]
```diff
@@ -1,42 +1,44 @@
...
+#  - bundles/shipment-table-rate
+#  - bundles/shipment-table-rate-data-import
+#  - bundles/splittable-checkout
+#  - bundles/splittable-totals
+#  - bundles/splittable-totals-rest-api
+#  - bundles/splittable-totals-rest-api-company-unit-address-connector
+#  - bundles/splittable-totals-rest-api-extension
+#  - bundles/splittable-totals-rest-api-shipment-connector
+#  - bundles/tax-calculation-connector
+#  - bundles/util-math-formula
```
Comment (daniel-rose) [inline codeception.yml:39]: Uncomment this lines.

File: composer.json [hunk 10]
```diff
@@ -175,6 +176,8 @@
       "FondOfOryx\\Zed\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode",
       "FondOfOryx\\Shared\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/src/FondOfOryx/Shared/PaymentEpcQrCode",
       "FondOfOryx\\Service\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode",
+      "FondOfOryx\\Service\\QrCodeGenerator\\": "bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator",
+      "FondOfOryx\\Shared\\QrCodeGenerator\\": "bundles/qr-code-generator/src/FondOfOryx/Shared/QrCodeGenerator",
```
Comment (daniel-rose) [inline composer.json:180]: Use alphabetically sort...

File: composer.json [hunk 11]
```diff
@@ -250,9 +253,11 @@
...
       "FondOfOryx\\Zed\\OneTimePasswordRestApi\\": "bundles/one-time-password-rest-api/tests/FondOfOryx/Zed/OneTimePasswordRestApi",
       "FondOfOryx\\Zed\\OneTimePassword\\": "bundles/one-time-password/tests/FondOfOryx/Zed/OneTimePassword",
-      "FondOfOryx\\Zed\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/src/FondOfOryx/Zed/PaymentEpcQrCode",
-      "FondOfOryx\\Shared\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/src/FondOfOryx/Shared/PaymentEpcQrCode",
-      "FondOfOryx\\Service\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode",
+      "FondOfOryx\\Zed\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/tests/FondOfOryx/Zed/PaymentEpcQrCode",
+      "FondOfOryx\\Shared\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/tests/FondOfOryx/Shared/PaymentEpcQrCode",
+      "FondOfOryx\\Service\\PaymentEpcQrCode\\": "bundles/payment-epc-qr-code/tests/FondOfOryx/Service/PaymentEpcQrCode",
+      "FondOfOryx\\Service\\QrCodeGenerator\\": "bundles/qr-code-generator/tests/FondOfOryx/Service/QrCodeGenerator",
+      "FondOfOryx\\Shared\\QrCodeGenerator\\": "bundles/qr-code-generator/tests/FondOfOryx/Shared/QrCodeGenerator",
```
Comment (daniel-rose) [inline composer.json:260]: Use alphabetically sort...

File: bundles/payment-epc-qr-code/composer.json (hunk 1 above)
Comment (daniel-rose) [reply bundles/payment-epc-qr-code/composer.json:13]: Look at dandelion.json

File: composer.json [hunk 12]
```diff
@@ -110,6 +110,7 @@
     "fond-of-oryx/splittable-totals-rest-api-shipment-connector": "*",
     "fond-of-oryx/tax-calculation-connector": "*",
     "fond-of-oryx/payment-epc-qr-code": "*",
+    "fond-of-oryx/qr-code-generator": "*",
```
Comment (daniel-rose) [inline composer.json:113]: Use alphabetically sort...

**Review by daniel-rose — CHANGES_REQUESTED** [review]

**Review by sourcecube — APPROVED** [review]

**Review by gengjozsef — APPROVED** [review]

**Review by daniel-rose — APPROVED** [review]

**Pull request merged**